rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/190_knn_query-with-k-param.yml

```diff
@@ -0,0 +1,204 @@
+        index: my_index
+        body:
+          size: 3
+          fields: [ my_name ]
+          query:
+            function_score:
+              query:
+                knn:
+                  field: my_vector
+                  query_vector: [ 1, 1, 1, 1 ]
+                  k : 5
+              functions:
+                - filter: { match: { my_name: v1 } }
+                  weight: 10
+                - filter: { match: { my_name: v2 } }
+                  weight: 100
+              boost_mode: multiply
+
+  - match: { hits.total.value: 5 } # collector sees k docs
+  - length: { hits.hits: 3 }
+  - match: { hits.hits.0._id: "2" }
+  - match: { hits.hits.0.fields.my_name.0: v2 }
+  - close_to: { hits.hits.0._score: { value: 50.0, error: 0.001 } }
+  - match: { hits.hits.1._id: "1" }
+  - match: { hits.hits.1.fields.my_name.0: v1 }
+  - close_to: { hits.hits.1._score: { value: 10.0, error: 0.001 } }
+  - match: { hits.hits.2._id: "4" }
+  - match: { hits.hits.2.fields.my_name.0: v2 }
+  - close_to: { hits.hits.2._score: { value: 10.0, error: 0.001 } }
+
+---
+"dis_max query with knn query":
+  - do:
+      search:
+        index: my_index
+        body:
+          size: 10
+          fields: [ my_name ]
+          query:
+            dis_max:
+              queries:
+                - knn: { field: my_vector, query_vector: [ 1, 1, 1, 1 ], k: 5, num_candidates: 10 }
+                - match: { my_name: v2 }
+              tie_breaker: 0.8
+
+  - match: { hits.total.value: 8 } # 5 knn results + extra results from match query
+  - match: { hits.hits.0._id: "2" }
+  - match: { hits.hits.0.fields.my_name.0: v2 }
+  - match: { hits.hits.1._id: "1" }
+  - match: { hits.hits.1.fields.my_name.0: v1 }
+  - match: { hits.hits.2._id: "4" }
+  - match: { hits.hits.2.fields.my_name.0: v2 }
+  - match: { hits.hits.3._id: "6" }
+  - match: { hits.hits.3.fields.my_name.0: v2 }
+  - match: { hits.hits.4._id: "8" }
+  - match: { hits.hits.4.fields.my_name.0: v2 }
+  - match: { hits.hits.5._id: "10" }
+  - match: { hits.hits.5.fields.my_name.0: v2 }
+  - match: { hits.hits.6._id: "3" }
+  - match: { hits.hits.6.fields.my_name.0: v1 }
+  - match: { hits.hits.7._id: "5" }
+  - match: { hits.hits.7.fields.my_name.0: v1 }
+
+---
+"Aggregations with collected number of docs depends on k param":
+  - do:
+      search:
+        index: my_index
+        body:
+          size: 2
+          query:
+            knn:
+              field: my_vector
+              query_vector: [1, 1, 1, 1]
+              k: 5 # collect 5 results from each shard
+          aggs:
+            my_agg:
+              terms:
+                field: my_name
+                order:
+                  _key: asc
+
+  - length: {hits.hits: 2}
+  - match: {hits.total.value: 5}
+  - match: {aggregations.my_agg.buckets.0.key: 'v1'}
+  - match: {aggregations.my_agg.buckets.1.key: 'v2'}
+  - match: {aggregations.my_agg.buckets.0.doc_count: 3}
+  - match: {aggregations.my_agg.buckets.1.doc_count: 2}
```

Comment:
Could you add some tests for the bad input case? `k=0` and `k < num_candidates`?

Reply:
Thanks for the review, addressed in 7f70a316d4af7585e1b460b26bbf2924ac0a55b4